### PR TITLE
[wfs] Fix MultiSurface and MultiCurve type detection, QGIS flavoured

### PR DESCRIPTION
**Pull request by m-kuhn**
QGIS server returns `gml:MultiSurfaceType` as and `gml:MultiCurveType` geometry type. Whereas for other geometry types it returns them suffixed with [Geom]**Property**Type.
Therefore the QGIS WFS provider client fails to detect the geometry types for those layers.

This inconsistency was introduced in https://github.com/qgis/QGIS/pull/36352 I'll open another pull request to align those with the rest.

Questions:

 - was this inconsistency (?) introduced on purpose and should be left as is, so only this (client, provider) patch merged?
 - was this not done on purpose and we should merge the other PR instead?
   - in this case, do we need to patch the wfs provider (client) with this pull request to make it working with the servers that have been released in the last couple of months?

In short, I wonder if we should patch client, server or both. I have a slight preference for server only, but that depends if there is infrastructure out there which already depends on the status quo.

@elpaso @rldhont I'd be keen to hear your thoughts.

### CODE REVIEW

**Comment by rldhont** [comment]
When I wrote #36352 I found this line in `geometryAggregates.xsd`:
```xml
<element name="MultiCurve" type="gml:MultiCurveType" substitutionGroup="gml:_GeometricAggregate"/>
```
So I supposed that `QgsWkbTypes::MultiCurve` has to be `gml:MultiCurveType`

I am not an xml schema warrior, so I don't know if the schema produced by QGIS Server has to contain `gml:MultiCurveType` or `gml:MultiCurvePropertyType`.

**Comment by m-kuhn** [comment]
If no-one objects and insists that this pull request should be merged, I'd say we close it without merging and suggest to upgrade the servers instead of adding extra bloat to the client.

**Comment by nyalldawson** [comment]
> If no-one objects and insists that this pull request should be merged, I'd say we close it without merging and suggest to upgrade the servers instead of adding extra bloat to the client.

Agreed
